### ISSUE TITLE
taint: add @psalm-flow for `Cache::remember`/`sear`/`flexible` (taint propagation through callbacks)

### ISSUE THREAD

**Issue by alies-dev** [psalm/psalm-plugin-laravel#588]
## Problem

`Cache::remember()`, `rememberForever()`, `sear()`, and `flexible()` accept a callback whose return value becomes the cached (and returned) value. Without `@psalm-flow`, taint is lost:

```php
// Taint lost — Psalm doesn't know the callback return flows through
$value = Cache::remember('key', 60, fn() => $request->input('q'));
echo $value; // No TaintedHtml — false negative
```

## Proposed fix

Add flow annotations to `stubs/common/Cache/Repository.stubphp`:

```php
namespace Illuminate\Cache;

class Repository
{
    /** @psalm-flow ($callback) -> return */
    public function remember($key, $ttl, Closure $callback) {}

    /** @psalm-flow ($callback) -> return */
    public function rememberForever($key, Closure $callback) {}

    /** @psalm-flow ($callback) -> return */
    public function sear($key, Closure $callback) {}

    /** @psalm-flow ($callback) -> return */
    public function flexible($key, $ttl, Closure $callback) {}
}
```

## Why not taint-source on get/pull/many?

See `docs/contribute/decisions.md` — "No taint-source on internal persistence reads". Cache reads return trusted data in 95%+ of cases; marking them as sources causes alert fatigue. Flow annotations catch the real danger (user input flowing through cache in the same analysis pass) without false positives.

Split from #545.

**Comment by alies-dev** [comment]
Converted to the repository internal ideas list (`.alias/issues.md`) so GitHub issues stay focused on active work. Reopen or recreate this when it is prioritized.

**Comment by alies-dev** [comment]
Converted to the repository internal ideas list (`.alias/issues.md`) so GitHub issues stay focused on active work. Reopen or recreate this when it is prioritized.
